.github/workflows/test-workflow.yml

```diff
@@ -25,22 +25,47 @@ jobs:
         aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
         aws-region: us-west-2
 
-    - name: Render Amazon ECS task definition
-      id: render-task-definition
-      uses: aws-actions/amazon-ecs-render-task-definition@964c59d79b99c32ccbda305c9e6afb0cec542076
+    - name: Render Amazon ECS task definition with task definition file 
+      id: render-task-definition-file
+      uses: aws-actions/amazon-ecs-render-task-definition@e246ecdd89c8de5a4a8c248830566528cbed06dd
       with:
         task-definition: task-definition.json
         container-name: sample-app
         image: amazon/amazon-ecs-sample:latest
 
-    - name: Validate Amazon ECS task definition
+    - name: Render Amazon ECS task definition with task definition arn (amazon resource name)
+      id: render-task-definition-arn
+      uses: aws-actions/amazon-ecs-render-task-definition@e246ecdd89c8de5a4a8c248830566528cbed06dd
+      with:
+        task-definition-arn: task-definition-arn
+        container-name: sample-app
+        image: amazon/amazon-ecs-sample:latest
+
+    - name: Render Amazon ECS task definition with task definition family 
+      id: render-task-definition-family
+      uses: aws-actions/amazon-ecs-render-task-definition@e246ecdd89c8de5a4a8c248830566528cbed06dd
+      with:
+        task-definition-family: fetch-def-task-definition
+        container-name: sample-app
+        image: amazon/amazon-ecs-sample:latest
+
+    - name: Render Amazon ECS task definition with task definition family and revision
+      id: render-task-definition-family-revision
+      uses: aws-actions/amazon-ecs-render-task-definition@e246ecdd89c8de5a4a8c248830566528cbed06dd
+      with:
+        task-definition-family: fetch-def-task-definition
+        task-definition-revision: 1
+        container-name: sample-app
+        image: amazon/amazon-ecs-sample:latest
+
+    - name: Validate Amazon ECS task definition file 
```

Comment:
Let's Keep the Validate and Register Steps together with the Render Step for file. So the order would look like:
1. Render for file
2. Validate for file
3. Register for file
4 .. and other renders.